.github/workflows/cleanup-artifacts.yml

```diff
@@ -0,0 +1,55 @@
+name: Cleanup Artifacts
+
+on:
+  workflow_dispatch:
+    
+  schedule:
+    - cron: '30 4 * * 5'
+            
+jobs:
+    jfrog:
+        name: "Cleanup Jfrog hpccpl-docker-local"
+        runs-on: ubuntu-22.04
+        if: ${{ github.repository == 'hpcc-systems/HPCC-Platform' }}
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                submodules: false
+            - name: Login to JFrog
+              uses: docker/login-action@v3
+              with:
+                registry: ${{ secrets.JFROG_REGISTRY }}
+                username: ${{ secrets.JFROG_USERNAME }}
+                password: ${{ secrets.JFROG_PASSWORD }}
+            - name: Setup JFrog CLI
+              uses: jfrog/setup-jfrog-cli@v3
+              with:
+                version: 2.56.0
+              env:
+                JF_URL: https://${{ secrets.JFROG_REGISTRY }}
+                JF_USER: ${{ secrets.JFROG_USERNAME }}
+                JF_PASSWORD: ${{ secrets.JFROG_PASSWORD }}
+
+            - name: 'Cleanup release-candidates over 1 month old'
+              shell: bash
+              run: |
+                jf rt del --quiet --spec ${{ github.workspace }}/dockerfiles/rule-hpccpl-docker-local-rcs-over-1mo.spec |& tee -a rc-deletion.log
+
+            - name: 'Cleanup point releases over 6 months old'
+              shell: bash
+              run: |
+                declare -a versions=("8.6" "8.8" "8.10" "8.12" "9.0" "9.2" "9.4" "9.6")
+                for version in "${versions[@]}"
+                do
+                  jf rt del --quiet --spec-vars "key1=${version}" --spec ${{ github.workspace }}/dockerfiles/rule-hpccpl-docker-local-point-over-6mo.spec --sort-by=created --sort-order=desc --offset=1 |& tee -a point-deletion-${version}.log
```

Comment:
Each version is done individually because the sortBy, sortOrder and offset functionality doesn't actually seem to be working in spec files. The sort, and order are necessary for 'offset' to drop the most recently created directory in the repo off the stack.  This helps to ensure that we always will retain at least one gold release for 'very' old versions.

For newer versions, I'll be implementing the {major.minor}-latest tags, which will also be excluded from these results.

Reply:
Also worth adding this as a comment - particularly --offset to ensure we always keep the last.